File: 강의 자료/02-알고리즘/03~04-정렬/문제별 코드/11652-카드/solution.py
import sys
n = int(sys.stdin.readline())
a = []

for i in range(n):
    a.append(int(sys.stdin.readline()))

a.sort()

mode = a[0]
modeCnt = 1
curCnt = 1

for i in range(1, n):
    if a[i] == a[i-1]:
        curCnt += 1
    else:
        curCnt = 1
    if modeCnt < curCnt:
        modeCnt = curCnt;
        mode = a[i]

print(mode)

File: 강의 자료/02-알고리즘/03~04-정렬/문제별 코드/11652-카드/solution.cpp
#include <iostream>
#include <algorithm>
typedef long long int ll;
using namespace std;

#define NM 100005

int N;
ll a[NM];

void input() {
	cin >> N;
	for (int i = 1; i <= N; i++) {
		cin >> a[i];
	}
}

void pro() {
    sort(a+1, a+1+N);
	ll mode = a[1];
    int modeCnt = 1, curCnt = 1;
	for (int i = 2; i <= N; i++) {
		if (a[i] == a[i - 1]) curCnt++;
		else curCnt = 1;
		if (modeCnt < curCnt) {
			modeCnt = curCnt;
			mode = a[i];
		}
	}
	cout << mode;
}


int main() {
	ios::sync_with_stdio(false); cin.tie(0); cout.tie(0);
	input();
	pro();
	return 0;
}